cmd/minishift/cmd/util/util_test.go

```diff
@@ -0,0 +1,42 @@
+/*
+Copyright (C) 2017 Red Hat, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package util
+
+import (
+	"testing"
+)
+
+func TestIsValidProfileName(t *testing.T) {
+	testData := []struct {
+		profileName string
+		expected    bool
+	}{{"test", true},
+		{".", false},
+		{"@foo", false},
+		{"foo123", true},
+		{"123foo", true},
+		{"foo_123", true},
+		{"foo@123", false},
+		{"", false},
+		{"foo-123", true}}
```

Comment:
We need to add `--foo` or `-foo123` as invalid profile names as `-` should be only used as separator.